pro_tes/ga4gh/tes/task_runs.py

```diff
@@ -366,10 +368,17 @@ def cancel_task(self, id: str, **kwargs) -> dict:
                 f"{db_document.tes_endpoint.host.rstrip('/')}/"
                 f"{db_document.tes_endpoint.base_path.strip('/')}"
             )
-            if self.store_logs:
-                task_id = db_document.task.logs[0].metadata.forwarded_to.id
+
+            assert db_document.task.logs is not None
+            logs: list[TesTaskLog] = db_document.task.logs
+
+            if (self.store_logs and logs and logs[0].metadata and
+                    logs[0].metadata.forwarded_to):
+                task_id = logs[0].metadata.forwarded_to.id
+                task_id = logs[0].metadata.forwarded_to.id
             else:
-                task_id = db_document.task.logs[0].metadata["remote_task_id"]
+                task_id = logs[0].metadata["remote_task_id"]    # type: ignore
```

Comment:
```
    metadata: Optional[Metadata] = Field(
        None,
        description=(
            "Arbitrary logging metadata included by the implementation."
        ),
        examples=[{"host": "worker-001", "slurmm_id": 123456}],
    )
```

```
class Metadata(CustomBaseModel):
    """Create model instance for metadata."""

    forwarded_to: Optional[TesNextTes] = Field(
        None,
        description="TaskLog describes logging information related to a Task",
    )
```
Clearly `metadata` is *not iterable*, so I don't know what the implementation is trying to do here.

Reply:
I think it actually _is_ iterable, or at least it's supposed to be. It is so that a task can be traced all the way to the end if being relayed by multiple gateways.